ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -454,14 +467,25 @@ public SchemaKStream<K> join() {
               contextStacker
           );
         case INNER:
-          return leftStream.innerJoin(
-              rightStream,
-              joinNode.getKeyColumnName(),
-              joinNode.withinExpression.get(),
-              JoiningNode.getValueFormatForSource(joinNode.left).getFormatInfo(),
-              JoiningNode.getValueFormatForSource(joinNode.right).getFormatInfo(),
-              contextStacker
-          );
+          if (joinNode.isSelfJoin) {
+            return leftStream.selfJoin(
```

Comment:
Did `selfJoin` get added to the DSL?

Reply:
Which DSL? The streams? No but this code is not using Streams DSL. It is building the ksql physical plan which will get translated to a streams topology in a later step